### PR TITLE
Add spring-cloud-deployer-spi-test-app as testapp

### LINKED ISSUE #581: Add spring-cloud-deployer-spi-test-app as test binary
Similarly we have testing packages for ticktock, log and helloworld, add `spring-cloud-deployer-spi-test-app` with different setting so that it's easier to play and test with apps which fail to start.

### PR DESCRIPTION
**Pull request by jvalkeal**
- Adding new packages for testapp using spring-cloud-deployer-spi-test-app
  which allows to play and test with failing app.
- 1.0.0 doesn't fail.
- 1.1.0 sets kill-delay=0 so app fails immediately.
- 1.2.0 sets kill-delay=30000 so app fails after 30s.
- Fixes #581

### CODE REVIEW

**Comment by ilayaperumalg** [comment]
@jvalkeal Could you look at the build failure from travis? Looks like there is an error with the PackageDocumentation test.

File: spring-cloud-skipper-server-core/src/test/resources/repositories/binaries/test/index.yml [hunk 1]
```diff
@@ -94,3 +94,39 @@ packageHomeUrl: http://cloud.spring.io/spring-cloud-stream-app-starters/
...
+apiVersion: skipper.spring.io/v1
+kind: SkipperPackageMetadata
+origin: samples-package-repository
+name: testapp
+version: 1.0.0
+packageSourceUrl: https://github.com/spring-cloud/spring-cloud-deployer
+packageHomeUrl: https://cloud.spring.io/spring-cloud-dataflow/
+tags: logging, sink
+maintainer: https://github.com/jvalkeal
+description: The testapp up starts and conditionally fails.
```
Comment (ilayaperumalg) [inline spring-cloud-skipper-server-core/src/test/resources/repositories/binaries/test/index.yml]: I guess this doesn't fail. will update the description when merging.

Comment (jvalkeal) [reply spring-cloud-skipper-server-core/src/test/resources/repositories/binaries/test/index.yml]: Yeah, those are just bogus field values. Or afaik not actually use anywhere.

**Comment by ilayaperumalg** [comment]
LGTM, verified the changes and merging.

**Comment by ilayaperumalg** [comment]
Squashed, rebased and merged as 07987b4

**Pull request closed without merging**